tests/resources/test_collection.py

```diff
@@ -303,3 +303,147 @@ async def test_get_collections_search(
+    assert len(cols) == 2
+    assert cols[0]["id"] == load_test_collection["id"]
+    assert cols[1]["id"] == load_test2_collection.id
+    assert len(links) == 2
+    assert {"root", "self"} == {link["rel"] for link in links}
+
+    ###################
+    # offset=3, because there are 2 collections, we should not have `next` or `prev` links
+    resp = await app_client.get(
+        "/collections",
+        params={"offset": 3},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 0
+    assert len(links) == 2
+    assert {"root", "self"} == {link["rel"] for link in links}
+
+    ###################
+    # offset=3,limit=1
+    resp = await app_client.get(
+        "/collections",
+        params={"limit": 1, "offset": 3},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 0
+    assert len(links) == 3
+    assert {"root", "self", "previous"} == {link["rel"] for link in links}
+    prev_link = list(filter(lambda link: link["rel"] == "previous", links))[0]
+    assert prev_link["href"].endswith("?limit=1&offset=2")
+
+    ###################
+    # limit=2, offset=3, there should not be a next link
+    resp = await app_client.get(
+        "/collections",
+        params={"limit": 2, "offset": 3},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 0
+    assert len(links) == 3
+    assert {"root", "self", "previous"} == {link["rel"] for link in links}
+    prev_link = list(filter(lambda link: link["rel"] == "previous", links))[0]
+    assert prev_link["href"].endswith("?limit=2&offset=1")
+
+    ###################
+    # offset=1,limit=1 should have a `previous` link
+    resp = await app_client.get(
+        "/collections",
+        params={"offset": 1, "limit": 1},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 1
+    assert cols[0]["id"] == load_test2_collection.id
+    assert len(links) == 3
+    assert {"root", "self", "previous"} == {link["rel"] for link in links}
+    prev_link = list(filter(lambda link: link["rel"] == "previous", links))[0]
+    assert "offset" not in prev_link["href"]
+
+    ###################
+    # BUG: pgstac doesn't return a `prev` link when limit is not set
+    # offset=1, should have a `previous` link
+    # resp = await app_client.get(
+    #     "/collections",
+    #     params={"offset": 1},
+    # )
+    # cols = resp.json()["collections"]
+    # links = resp.json()["links"]
+    # assert len(cols) == 1
+    # assert cols[0]["id"] == load_test2_collection.id
+    # assert len(links) == 3
+    # assert {"root", "self", "previous"} == {link["rel"] for link in links}
+    # prev_link = list(filter(lambda link: link["rel"] == "previous", links))[0]
+    # # offset=0 should not be in the previous link (because it's useless)
+    # assert "offset" not in prev_link["href"]
```

Comment:
I believe this is a PgSTAC issue

For some reason when `offset=1` and `limit` is not set, we don't get `prev` link (it should because there are 2 collections in the db

cc @bitner

Reply:
🤔 maybe it makes sense not to have a link but I'm sure we could dynamically create one with `offset=0`